src/components/Carousel/Carousel.js

```diff
@@ -103,121 +74,58 @@ class Carousel extends Component {
-
-    if (nextActiveIndex < 0) {
-      this.setActivePanel(panels.length - 1);
-    } else {
-      this.setActivePanel(nextActiveIndex);
-    }
-  }
-
-  renderAutoPlay() {
-    const { autoPlayStartText, autoPlayStopText } = this.props;
-    const { autoPlayActive } = this.state;
-
-    const icon = autoPlayActive ? 'pause' : 'right';
-    const assistiveText = autoPlayActive
-      ? autoPlayStopText
-      : autoPlayStartText;
-
-    return (
-      <span className="slds-carousel__autoplay">
-        <IconButton
-          icon={icon}
-          border="filled"
-          sprite="utility"
-          onClick={this.toggleAutoPlay}
-          size="x-small"
-          title={assistiveText}
-        />
-      </span>
-    );
+  startAutoPlay() {
+    const { autoPlayInterval } = this.props;
+    this.autoPlayId = setInterval(this.activateNextPanel, autoPlayInterval);
+    this.setState({ autoPlayActive: true });
   }
 
-  renderIndicator = ({ props: { active, id, title } }, index) => {
-    const tabIndex = active ? 0 : -1;
-
-    const actionClasses = cx(
-      'slds-carousel__indicator-action',
-      { 'slds-is-active': active },
-    );
-
-    return (
-      <li className="slds-carousel__indicator" key={id} role="presentation">
-        <a
-          className={actionClasses}
-          role="tab"
-          tabIndex={tabIndex}
-          aria-selected={active}
-          aria-controls={id}
-          onClick={this.onClickIndicator}
-          title={title}
-          data-index={index}
-        >
-          <span className="slds-assistive-text">{title}</span>
-        </a>
-      </li>
-    );
+  stopAutoPlay() {
+    clearInterval(this.autoPlayId);
+    this.autoPlayId = null;
+    this.setState({ autoPlayActive: false });
   }
 
   render() {
     const {
       autoPlay,
+      autoPlayStartText,
+      autoPlayStopText,
       className,
+      children,
     } = this.props;
 
     const {
       activeIndex,
-      panels,
+      autoPlayActive,
     } = this.state;
 
-    const sldsClasses = [
-      'slds-carousel',
-      className
-    ];
-
-    const translateX = `translateX(-${activeIndex * 100}%)`;
-
     return (
-      <div className={cx(sldsClasses)}>
-        <div className="slds-carousel__stage">
-          {autoPlay && this.renderAutoPlay()}
-          <div className="slds-carousel__panels" style={{ transform: translateX }}>
-            {panels}
-          </div>
-          <ul className="slds-carousel__indicators" role="tablist">
-            {panels.map(this.renderIndicator)}
-          </ul>
-        </div>
-      </div>
+      <ControlledCarousel
+        activeIndex={activeIndex}
+        autoPlay={autoPlay}
+        autoPlayActive={autoPlayActive}
+        autoPlayStartText={autoPlayStartText}
+        autoPlayStopText={autoPlayStopText}
+        className={className}
+        onToggleAutoPlay={() => this.toggleAutoPlay()}
+        onPanelChange={index => this.setState({ activeIndex: index })}
```

Comment:
Should be moved to a method, e.g. `this.panelChange` and `panelChange = (index) => {}`